### PR TITLE
Update docker logs collection with AD label supporting file tailing

### PR DESCRIPTION
**Pull request by prognant**
### What does this PR do?
Document new feature coming with Agent 7.25.0 (https://github.com/DataDog/datadog-agent/pull/6883).

### Motivation
Keep doc up-to-date.

### Preview
https://docs-staging.datadoghq.com/prognant/file-tailing-from-docker-AD-label/agent/docker/log/

### Additional Notes
It will be available for agent >= 7.25 (will be available around the beginning of the year)

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/docker/log.md [hunk 1]
```diff
@@ -228,6 +228,31 @@ See the [multi-line processing rule documentation][1] to get more pattern exampl
...
+labels:
+    com.datadoghq.ad.logs: '[{"type":"file", "source": "sample_app", "service": "sample_service", "path": "/logs/app/prod.log"}]'
+```
+
+Log collected from that file will be tagged with the container metadata. Log collection is linked to the container life cycle, as soon as the container stops, log collection from that file also stops.
+
+
+**Important notes**:
+
+- The path is **relative** to the Agent, thus the directory containing the file should be shared between the container running the application logging to the file and the Agent container. For example the container may mount `/logs` and each container logging to file may mount a volume like `/logs/app` where the log file will be written.
```
Comment (ruthnaebeck) [inline content/en/agent/docker/log.md:246]:
```suggestion
- The file path is **relative** to the Agent, so the directory containing the file should be shared between the container running the application and the Agent container. For example, if the container mounts `/logs` each container logging to file may mount a volume such as `/logs/app` where the log file is written.
```

File: content/en/agent/docker/log.md [hunk 2]
```diff
@@ -228,6 +228,31 @@ See the [multi-line processing rule documentation][1] to get more pattern exampl
...
+
+- When using this kind of label on a container, its `stderr`/`stdout` logs are not collected automatically. If collection from both `stderr`/`stdout` and a file are needed it should be explicity enabled by using a label like:
```
Comment (ruthnaebeck) [inline content/en/agent/docker/log.md:248]:
```suggestion
- When using this kind of label on a container, its `stderr`/`stdout` logs are not collected automatically. If collection from both `stderr`/`stdout` and a file are needed it should be explicity enabled by using a label, for example:
```

File: content/en/agent/docker/log.md [hunk 3]
```diff
@@ -228,6 +228,31 @@ See the [multi-line processing rule documentation][1] to get more pattern exampl
...
+```yaml
+labels:
+    com.datadoghq.ad.logs: '[{"type":"file", "source": "java", "service": "app", "path": "/logs/app/prod.log"}, {"type": "docker", "source": "app_container", "service": "app"}]'
+```
+
+- When using this kind of combination, `source` and `service` have no default value, and therefore they should be explicitely set in the Autodiscovery label.
```
Comment (ruthnaebeck) [inline content/en/agent/docker/log.md:254]:
```suggestion
- When using this kind of combination, `source` and `service` have no default value and should be explicitly set in the Autodiscovery label.   
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]

**Comment by sarina-dd** [comment]
@cohenyair Did a review of the content and this looks great! 👍 I'll let Ruth finalize the review as she requested previous changes during the editorial review.

**Comment by sarina-dd** [comment]
@prognant Hi there! Checking in on status: are we okay to merge this now that the feature is released, or are we waiting on additional merges?

**Comment by prognant** [comment]
Hi @sarina-dd, the agent release (7.25.0) targeted by this documentation PR is due for release soon (unsure exactly when), If that's ok I can ping you here as soon as we got 7.25.0 out ?

**Comment by sarina-dd** [comment]
@prognant Yep, sounds good 👍 Just ping me whenever we're ready and I can publish. Thanks for this!

**Comment by prognant** [comment]
Hi @sarina-dd, the [Agent v7.25.0 ](https://github.com/DataDog/datadog-agent/releases/tag/7.25.0) so you can now merge this PR !

**Review by sarina-dd — APPROVED** [review]

**Review by ruthnaebeck — APPROVED** [review]
👍

**Pull request merged**